### ISSUE TITLE
Chart keep displaying data if no data is found in new request.

### ISSUE THREAD

**Issue by mrfigueiredo** [michaeldmoore/michaeldmoore-multistat-panel#58]
If i make a query that has data in the result, the chart will update.
But after that i change to a time interval that return no data, the chart does not update, it still shows the result from last query.
Is that the expected behavior ?

**Comment by michaeldmoore** [comment]
No.  If the new query returns no data, it should display a message saying
'no data'

Check the versions on Grafana and multistat and make sure you are all up to
date, then if it looks good, look at the query inspector and see what data
is really being received.

Let me know what you find, and if you still have problems, send some
details, including l, if possible, the panel JSON and the query inspector
JSON.

**Comment by felipetomm** [comment]
Hello @michaeldmoore,
as reported by @mrfigueiredo (we are working together), our Grafana is v7.1.5 and Multistat plugin is 1.4.1.  

This is panel JSON: [JSON-Panel.txt](https://github.com/michaeldmoore/michaeldmoore-multistat-panel/files/5423605/JSON-Panel.txt)  
When has no data, that's the response: [JSON-Multistat-HasNoData.txt](https://github.com/michaeldmoore/michaeldmoore-multistat-panel/files/5423591/JSON-Multistat-HasNoData.txt)  
And this is the response when has data: [JSON-Multistat-HasData.txt](https://github.com/michaeldmoore/michaeldmoore-multistat-panel/files/5423645/JSON-Multistat-HasData.txt)  

With that query and same filters, others plugins showing 'no data' (or similar).

**Comment by michaeldmoore** [comment]
Very interesting.

I can't exactly reproduce it my end - I'd need to install Mongo and somehow
recreate the schema and data you are using.  However, from the two response
samples you gave, I think I'm on to something.

When there is data in the time interval, you're receiving this:
"response": [
    {
      "columns": [
         { "text": "_id", "type": "text" },
         { "text": "trf", "type": "text" }
      ],
      "rows": [
        [ "Suporte", 3],
        [ "IT", 1]
      ],
      "type": "table"
    }

This looks fine, and as you say - it works.

When you narrow the time interval down to a point where there is no data
though, you're receiving this:
  "response": [
    {
      "columns": [],
      "rows": [],
      "type": "table"
    }

My guess is that Multistat is crapping out when there are no column
definitions and failing to generate the appropriate "no data" display.


This shouldn't be hard to find and fix.  I'll get to this tomorrow.

Thanks for the report and stay turned for a follow-up email once the fix is
in.
Michael Moore

**Comment by michaeldmoore** [comment]
OK, I have a quick fix you should be able to apply yourself without having
to wait for an official release.

Look in the module.js file (in the
plugins/michaeldmoore-multistat-panel directory (or in the dist folder, if
there is one in there)
Open this with a simple text editor and find the onDataReceived function
(should be around line 262 or so)
Add in the text shown below in yellow, and save it.
[image: image.png]

Now, whenever the rows array is empty, it should display the "No data to
show" message.

This will be part of the next official release.  Thank you for pointing out
the problem.
Regards, and let me know if you still have problems after this,
Michael Moore

**Comment by felipetomm** [comment]
@michaeldmoore, image attach was not uploaded. Can you resend it?
